File: examples/pizzas_domain/lib/pizzas_domain/order/events/canceled_order.rb
module PizzasDomain
  class Order
    module Events
      class CanceledOrder
        attr_reader :aggregate_id, :order, :customer_name, :items, :status, :occurred_at

        def initialize(aggregate_id: nil, order: nil, customer_name: nil, items: nil, status: nil)
          @aggregate_id = aggregate_id
          @order = order
          @customer_name = customer_name
          @items = items
          @status = status
          @occurred_at = Time.now
          freeze
        end
      end
    end
  end
end

File: pizzas_static_go/domain/canceled_order.go
package domain

import "time"

type CanceledOrder struct {
	AggregateID string `json:"aggregate_id"`
	AggregateId string `json:"aggregate_id"`
	CustomerName string `json:"customer_name"`
	Items []OrderItem `json:"items"`
	Status string `json:"status"`
	OccurredAt time.Time `json:"occurred_at"`
}

func (c CanceledOrder) EventName() string { return "CanceledOrder" }

func (c CanceledOrder) GetOccurredAt() time.Time { return e.OccurredAt }
